tools/pipelines-tasks/AVDAppAttachPublish/main.ts

```diff
@@ -0,0 +1,131 @@
+
+		let subscriptionID: string = tl.getEndpointDataParameter(connectedServiceName, 'subscriptionid', true)!;
+
+		let storageAccountName = tl.getInput('storageAccount', true)!;
+		let azureEndpoint: AzureEndpoint = await new AzureRMEndpoint(connectedServiceName).getEndpoint();
+		let accessToken: string = (await azureEndpoint.applicationTokenCredentials.getToken()).toString();
+
+		const storageArmClient = new armStorage.StorageManagementClient(azureEndpoint.applicationTokenCredentials, (azureEndpoint.subscriptionID)!);
+		let storageAccount: StorageAccount = await storageArmClient.storageAccounts.get(storageAccountName);
+		let storageAccountResourceGroupName = getResourceGroupNameFromUri(storageAccount.id);
+
+		let accessKeys = await storageArmClient.storageAccounts.listKeys(storageAccountResourceGroupName, storageAccountName, null);
+		let accessKey: string = accessKeys[0];
+		let storageAccountConnectionString: string = `DefaultEndpointsProtocol=https;AccountName=${storageAccountName};AccountKey=${accessKey};EndpointSuffix=core.windows.net`;
+		let username: string = (process.env.BUILD_REQUESTEDFOREMAIL)!;
+		const PUBLISH_TO_AZURE_TASK = '8';
+
+		const appattachConfig = {
+			'emailId': username,
+			'accessToken': accessToken,
+			'azureStorageKey': storageAccountConnectionString,
+			'azureWorkspace': tl.getInput('workSpace', true),
+			'azureStorageFileShare': tl.getInput('fileShare', true),
+			'azureStorageAccount': storageAccountName,
+			'azureSubscriptionId': subscriptionID,
+			'azureResourceGroup': tl.getInput('resourceGroupName', true),
+			'azureHostPoolName': tl.getInput('hostPool', true),
+			'azureApplicationGroupName': tl.getInput('applicationGroup', true),
+			'taskType': PUBLISH_TO_AZURE_TASK,
+			'appAttachImagePath': tl.getInput('vhdxPath', true),
+			'clientType': helpers.CLIENT_TYPE,
+			'clientVersion': helpers.CLIENT_VERSION
+		};
+
+		const jsonString = JSON.stringify(appattachConfig);
+		tl.debug(jsonString);
+
+		const powershellRunner: ToolRunner = helpers.getPowershellRunner(HELPER_SCRIPT);
+		powershellRunner.arg(['-inputJsonStr', '\'' + jsonString + '\'']);
+		powershellRunner.arg(['-targetDLL', TARGET_DLL]);
+
+		const execResult = await powershellRunner.execSync();
+
+		if (execResult.code) {
+			throw execResult.stderr;
+		}
+
+		// display log file content to console
+		const appAttachLogDir: string = path.join(os.tmpdir(), 'AppAttach');
+		const appAttachLogDirFiles = fs.readdirSync(appAttachLogDir);
+		const appAttachlogFile = path.join(appAttachLogDir, appAttachLogDirFiles[0]);
+		fs.readFile(appAttachlogFile, 'utf8', (err, data) => {
+			if (err) {
+				console.error('Error reading the file:', err);
+			} else {
+				console.log(appAttachlogFile, data);
+			}
+		}); 
+	} catch (error) {
+		isSuccessful = false;
+		console.error(tl.loc("AppAttachPublish Error", error));
+		const powershellRunner: ToolRunner = helpers.getPowershellRunner(HELPER_SCRIPT_EXCEPTIONS_TELEMETRY);
```

Comment:
could this also be done through `telemetry.emitTelemetry`?

Reply:
Done. logging exception message through telemetry.emitTelemetry too